### PR TITLE
Add support for querying vertex attributes in reflection API

### PR DESCRIPTION
**Pull request by thp**
This adds support for querying which vertex attributes are specified as inputs in the vertex shader and for querying the number of attributes, their names and types similar to querying uniform variables.

### CODE REVIEW

File: glslang/MachineIndependent/reflection.cpp [hunk 1]
```diff
@@ -671,6 +687,10 @@ void TLiveTraverser::visitSymbol(TIntermSymbol* base)
 {
     if (base->getQualifier().storage == EvqUniform)
         addUniform(*base);
+
+    // FIXME: Is this the right way to distinguish vertex inputs from fragment inputs?
```
Comment (johnkslang) [inline glslang/MachineIndependent/reflection.cpp]: No, this is just asking if the inputs had smooth, flat, or noperspective qualifiers.

You should be able to tell the stage by looking at `language`; e.g., intermediate.getStage().

File: glslang/MachineIndependent/reflection.cpp [hunk 2]
```diff
@@ -671,6 +687,9 @@ void TLiveTraverser::visitSymbol(TIntermSymbol* base)
 {
     if (base->getQualifier().storage == EvqUniform)
         addUniform(*base);
+
+    if (intermediate.getStage() == EShLangVertex && base->getQualifier().isPipeInput())
```
Comment (thp) [inline glslang/MachineIndependent/reflection.cpp]: Tested this with some of my shaders and seems to work well, checking if the stage is the vertex language and then just checking if it's an input variable.

File: glslang/MachineIndependent/reflection.cpp [hunk 3]
```diff
@@ -728,6 +747,11 @@ void TReflection::dump()
         indexToUniformBlock[i].dump();
     printf("\n");
 
+    printf("Vertex attribute reflection:\n");
```
Comment (thp) [inline glslang/MachineIndependent/reflection.cpp]: Also added this block here to be able to look at the generated reflection data using:

```
build/StandAlone/glslangValidator -l -q shader.vert shader.frag
```

File: Test/reflection.vert [hunk 4]
```diff
@@ -94,6 +94,12 @@ struct deep3 {
     ivec3 v3;
 };
 
+in float attributeFloat;
```
Comment (thp) [inline Test/reflection.vert]: I tried to test different types and with/without layout qualifiers here, and with different types -- anything else that comes to mind that should be tested that isn't here already?

**Pull request merged**